runtime-sdk/modules/evm/src/lib.rs

```diff
@@ -415,20 +389,44 @@ impl<Cfg: Config> API for Module<Cfg> {
             tx_metadata,
         ) = Self::decode_simulate_call_query(ctx, call)?;
 
-        let call_tx = transaction::Transaction {
+        let (method, body) = match address {
```

Comment:
Could remember/store into a variable if this is a call or create, to avoid matching on address below again (and repeating the comment)

Reply:
Good idea, refactored using a closure, take a look.

Reply:
Looks good, thx!